packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1593,8 +1593,10 @@ export function restoreRootViewTransitionName(rootContainer: Container): void {
     // clone the whole document outside of the React too.
     containerInstance = (rootContainer: any);
   }
-  // $FlowFixMe[prop-missing]
-  if (containerInstance.style.viewTransitionName === 'root') {
+  if (
+    containerInstance.style &&
+    containerInstance.style.viewTransitionName === 'root'
+  ) {
```

Comment:
```suggestion
  if (
    (disableCommentsAsDOMContainers || containerInstance.nodeType !== COMMENT_NODE) &&
    // $FlowFixMe[prop-missing]
    containerInstance.style.viewTransitionName === 'root'
  ) {
```

Since most of this applying and restoring only applies to the documentElement which we get even if it's a comment. This code is mainly gated on the gesture case. So we can ignore it if it's not a gesture case.

However, I think we probably should throw in the gesture case if it's a comment container so that it throws before getting to the restore.

Reply:
We should throw here if it's a comment node container:

https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js#L1706-L1715